modules/robot/src/real-robot/jaco/vpRobotJaco2.cpp

```diff
@@ -0,0 +1,326 @@
+ * distribution for additional information about the GNU GPL.
+ *
+ * For using ViSP with software that can not be combined with the GNU
+ * GPL, please contact Inria about acquiring a ViSP Professional
+ * Edition License.
+ *
+ * See http://visp.inria.fr for more information.
+ *
+ * This software was developed at:
+ * Inria Rennes - Bretagne Atlantique
+ * Campus Universitaire de Beaulieu
+ * 35042 Rennes Cedex
+ * France
+ *
+ * If you have questions regarding the use of this file, please contact
+ * Inria at visp@inria.fr
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+ * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * Description:
+ * Interface for Kinova Jaco robot.
+ *
+ * Authors:
+ * Fabien Spindler
+ *
+ *****************************************************************************/
+
+#include <visp3/core/vpConfig.h>
+
+#ifdef VISP_HAVE_JACO2
+
+#include <visp3/robot/vpRobotException.h>
+
+/*!
+  \file vpRobotJaco2.cpp
+  \brief Interface for Kinova Jaco2 robot.
+*/
+
+#include <visp3/core/vpHomogeneousMatrix.h>
+#include <visp3/robot/vpRobotJaco2.h>
+
+/*!
+  Basic initialization.
+ */
+void vpRobotJaco2::init()
+{
+  // If you want to control the robot in Cartesian in a tool frame, set the corresponding transformation in m_eMc
+  // that is set to identity by default in the constructor.
+
+  maxRotationVelocity = maxRotationVelocityDefault;
+  maxTranslationVelocityDefault = maxTranslationVelocityDefault;
+
+  // Set here the robot degrees of freedom number
+  nDof = 6; // Jaco2 has 6 dof
+}
+
+/*!
+  Default constructor.
+ */
+vpRobotJaco2::vpRobotJaco2()
+  : m_eMc()
+{
+  init();
+}
+
+/*!
+  Destructor.
+ */
+vpRobotJaco2::~vpRobotJaco2()
+{
+  std::cout << "Not implemented ! " << std::endl;
+}
+
+/*
+
+  At least one of these function has to be implemented to control the robot with a
+  Cartesian velocity:
+  - get_eJe()
+  - get_fJe()
+
+*/
+
+/*!
+  Get the robot Jacobian expressed in the end-effector frame.
+
+  \param[out] eJe : End-effector frame Jacobian.
+*/
+void vpRobotJaco2::get_eJe(vpMatrix &eJe)
+{
+  (void) eJe;
+  std::cout << "Not implemented ! " << std::endl;
+}
+
+/*!
+  Get the robot Jacobian expressed in the robot reference frame.
+
+  \param[out] fJe : Base (or reference) frame Jacobian.
+*/
+void vpRobotJaco2::get_fJe(vpMatrix &fJe)
+{
+  (void) fJe;
+  std::cout << "Not implemented ! " << std::endl;
+}
+
+/*
+
+  At least one of these function has to be implemented to control the robot:
+  - setCartVelocity()
+  - setJointVelocity()
+
+*/
+
+/*!
+  Send to the controller a 6-dim velocity skew vector expressed in a Cartesian frame.
+
+  \param[in] frame : Cartesian control frame (either tool frame or end-effector) in which the velocity \e v is expressed.
+  Units are m/s for translation and rad/s for rotation velocities.
+
+  \param[in] v : 6-dim vector that contains the 6 components of the velocity skew to send to the robot.
```

Comment:
Wondering if `velocity skew` is the correct term?

`Velocity twist` looks more appropriate in my opinion ([Screw theory](https://en.wikipedia.org/wiki/Screw_theory)).

Reply:
You are completely right.
Fixed in my next commit